.github/docker/ubuntu-22.04.Dockerfile

```diff
@@ -0,0 +1,71 @@
+# Copyright (C) 2023 Intel Corporation
+# Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+# See LICENSE.TXT
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#
+# Dockerfile - image with all Unified Runtime dependencies.
+#
+
+# Pull base image
+FROM registry.hub.docker.com/library/ubuntu:22.04
+
+# Set environment variables
+ENV OS ubuntu
+ENV OS_VER 22.04
+ENV NOTTY 1
+ENV DEBIAN_FRONTEND noninteractive
+
+# Additional parameters to build docker without building components.
+# These ARGs can be set in docker building phase and are used
+# within bash scripts (executed within docker).
+ARG SKIP_DPCPP_BUILD
+ARG SKIP_LIBBACKTRACE_BUILD
+
+# Base development packages
+ARG BASE_DEPS="\
+	build-essential \
+	cmake \
+	git"
+
+# Unified Runtime's dependencies
+ARG UR_DEPS="\
+	doxygen \
+	python3 \
+	python3-pip"
+
+# Unified Runtime's dependencies (installed via pip)
+ARG UR_PYTHON_DEPS="\
+	clang-format==15.0.7"
+
+# Miscellaneous for our builds/CI (optional)
+ARG MISC_DEPS="\
+	clang \
+	sudo \
+	wget \
+	whois"
+
+# Update and install required packages
+RUN apt-get update \
+ && apt-get install -y --no-install-recommends \
+	${BASE_DEPS} \
+	${UR_DEPS} \
+	${MISC_DEPS} \
+ && apt-get clean all
+
+RUN pip3 install ${UR_PYTHON_DEPS}
+
+# Install DPC++
+COPY install_dpcpp.sh install_dpcpp.sh
+ENV DPCPP_PATH=/opt/dpcpp
+RUN ./install_dpcpp.sh
+
+# Install libbacktrace
+COPY install_libbacktrace.sh install_libbacktrace.sh
+RUN ./install_libbacktrace.sh
+
+# Add and switch to a new (non-root) 'user'
+ENV USER user
+ENV USERPASS pass
+RUN useradd -m $USER -g sudo -p `mkpasswd $USERPASS`
+USER $USER
```

Comment:
I personally use `podman` instead of `docker` which when used in rootless mode matches the uid/gid of the containers root user to that of the current host user which is nice when mounting volumes in the host user home directory as any modified files have the corrent owner. This is obviously not how `docker` works, since all containers are run as the host root user.

Having the container image setup a user breaks the ease of use of podman running in rootless mode. Would it be possible to make user creation optional somehow?

Reply:
Another approach I've seen with `docker` is to explicitly map the containers root user to the current host users uid/gid by passing `--user $UID:$GID` to `docker run`.

Reply:
It seems like a nice thing, but... I can't make it work locally (using `docker`). I guess it's time to switch to podman 😉 

Anyway, I removed the `USER` command, since we can switch to `user` in CI, using your proposed `--user` param.

Reply:
> It seems like a nice thing, but... I can't make it work locally (using docker).

That's a shame.

> Anyway, I removed the USER command, since we can switch to user in CI, using your proposed --user param.

Just tried out the Dockerfile with `podman` and its working nicely now. Thanks :)